crates/flashblocks-archiver/src/archiver.rs

```diff
@@ -0,0 +1,144 @@
+        );
+
+        if self.config.builders.is_empty() {
+            warn!("No builders configured, archiver will not collect any data");
+            return Ok(());
+        }
+
+        // Start WebSocket connections
+        let ws_pool = WebSocketPool::new(self.config.builders.clone());
+        let mut receiver = ws_pool.start().await?;
+
+        // Set up batching
+        let mut batch = Vec::with_capacity(self.config.archiver.batch_size);
+        let mut flush_interval = interval(Duration::from_secs(
+            self.config.archiver.flush_interval_seconds,
+        ));
+
+        info!("FlashblocksArchiver started, listening for flashblock messages");
+
+        loop {
+            tokio::select! {
+                // Receive flashblock messages
+                message = receiver.recv() => {
+                    match message {
+                        Some((builder_name, payload)) => {
+                            if let Some(builder_id) = self.builder_ids.get(&builder_name) {
+                                batch.push((*builder_id, payload));
+
+                                // Flush if batch is full
+                                if batch.len() >= self.config.archiver.batch_size {
+                                    if let Err(e) = self.flush_batch(&mut batch).await {
+                                        error!("Failed to flush batch: {}", e);
+                                        self.metrics.flush_batch_error.increment(1);
+                                    }
+                                }
+                            } else {
+                                warn!("Received message from unknown builder: {}", builder_name);
+                            }
+                        }
+                        None => {
+                            info!("All WebSocket connections closed, flushing remaining data");
+                            if !batch.is_empty() {
+                                if let Err(e) = self.flush_batch(&mut batch).await {
+                                    error!("Failed to flush final batch: {}", e);
+                                }
+                            }
+                            break;
+                        }
+                    }
+                }
+
+                // Periodic flush
+                _ = flush_interval.tick() => {
+                    if !batch.is_empty() {
+                        if let Err(e) = self.flush_batch(&mut batch).await {
+                            error!("Failed to flush batch on timer: {}", e);
+                        }
+                    }
+                }
+            }
+        }
+
+        info!("FlashblocksArchiver stopped");
+        Ok(())
+    }
+
+    async fn flush_batch(&self, batch: &mut Vec<(Uuid, FlashblockMessage)>) -> Result<()> {
+        if batch.is_empty() {
+            return Ok(());
+        }
+
+        info!("Flushing batch of {} flashblock messages", batch.len());
```

Comment:
Can we switch the logs to the structured logging? ([example](https://github.com/base/node-reth/blob/main/crates/flashblocks-rpc/src/state.rs#L350C21-L354))

Makes it much easier to work with/alert on once it has been ingested